homeassistant/components/hausbus/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The Haus-Bus integration."""
+from __future__ import annotations
+
+import logging
+from typing import cast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .gateway import HausbusGateway
+
+# Only support light platform on the initial version. Additional platforms to follow
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Haus-Bus from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    gateway = HausbusGateway(hass, entry)
+    hass.data[DOMAIN][entry.entry_id] = gateway
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    _LOGGER.debug("start searching devices")
+
+    # search devices after adding all callbacks to the gateway object
+    gateway.home_server.searchDevices()
```

Comment:
can't this throw exceptions? Can we do this before we setup the entries, so we abort if anything goes wrong?

Reply:
This does essentially only send a UDP broadcast into the network to trigger the device discovery. The actual devices will be populated by events that are received in response to this.